iotedgedev/azurecli.py

```diff
@@ -0,0 +1,150 @@
+import json
+from fstrings import f
+from io import StringIO
+from azure.cli.core import get_default_cli
+
+class AzureCli:
+    def __init__(self, envvars, output):
+        self.envvars = envvars
+        self.output = output
+        self.az_cli = get_default_cli()
+
+    def invoke_az_cli(self, args, error_message, io=None): 
+        try:
+            exit_code = self.az_cli.invoke(args, out_file=io)
+            if exit_code and exit_code != 0:
+                self.output.error(error_message)
+                return False
+        except Exception as e:
+            self.output.error(error_message)
+            self.output.error(str(e))
+            return False
+
+        self.output.footer("Success")
+
+        return True
+
+    def add_extension(self, name):
+        self.output.header(f("Adding extension {name}"))
+
+        return self.invoke_az_cli(["extension", "add", "--name",name,
+                                   "--yes"],
+                                  f("Error while adding extension {name}"))
+
+    def extension_exists(self, name):
+        self.output.header(f("Checking for extension {name}"))
+
+        return self.invoke_az_cli(["extension", "show", "--name", name, "--output", "table"],
+                                  f("Error while checking for extension {name}"))
+
+    def login(self, username, password):
+        self.output.header("Loging in to Azure")
+
+        return self.invoke_az_cli(["login", "-u", username,
+                                   "-p",  password, "--query", "\"[].[name, resourceGroup]\"", "-o", "table"],
+                                  "Error while trying to login to Azure")
+
+    def login_interactive(self):
+        self.output.header("Interactive login to Azure")
+
+        return self.invoke_az_cli(["login", "--query", "\"[].[name, resourceGroup]\"", "--o", "table"],
+                                  "Error while trying to login to Azure")
+
+    def list_subscriptions(self):
+        self.output.header("Listing Subscriptions")
+
+        return self.invoke_az_cli(["account", "list", "--out", "table"],
+                                  "Error while trying to list Azure subscriptions")
+
+    def set_subscription(self, subscription):
+        self.output.header("Setting Subscription")
+
+        return self.invoke_az_cli(["account", "set", "--subscription", subscription],
+                                  "Error while trying to set Azure subscription")
```

Comment:
Error messages should be consistent with ending period.  You have "." at the end of some and not others.